apache/metron
# METRON-1499 Enable Configuration of Unified Enrichment Topology via Ambari

We recently introduced an alternative enrichment topology; the Unified enrichment topology.  This is an alternative to the existing Split Join topology.  This work enhances the MPack so that a user can configure, stop, and start either the Unified or the Split Join enrichment topology.  

* The Split Join topology remains the default Enrichment topology.

* A combo box allows the user to choose which Enrichment topology should be run.

    ![screen shot 2018-04-05 at 5 41 52 pm](https://user-images.githubusercontent.com/2475409/38395683-27272e9a-3902-11e8-9be6-bbd3fbe0ee68.png)

* Each topology has a separate configuration section under the Metron > Enrichment settings in Ambari.

    ![screen shot 2018-04-05 at 5 42 19 pm](https://user-images.githubusercontent.com/2475409/38395690-3188795c-3902-11e8-8064-0288e424d142.png)

* The `bin/start_enrichment_topology.sh` script was enhanced so that you can pass in the `--filter` and `--remote` args directly to launch an alternative topology.  Running the script with no parameters will launch the default topology; Split Join.

* The Unified topology had previously reused some of the same settings from the Split Join topology.  Where it made sense, the Unified topology was updated to use its own logically named properties.

*  There are now separate properties file for each Enrichment topology. 
    `$METRON_HOME/config/enrichment-splitjoin.properties`
    `$METRON_HOME/config/enrichment-unified.properties`

* There are separate flux files for each Enrichment topology. 
    `$METRON_HOME/flux/enrichment/remote-splitjoin.yaml` 
    `$METRON_HOME/flux/enrichment/remote-unified.yaml`






### Manual Testing

To test this change, launch a development environment.  The Split Join Enrichment topology will be running by default.  Switch to the Unified topology.  Change some settings and make sure they are reflected in the topology.

## Pull Request Checklist
